### PR TITLE
Feature/dtserwthre 157 i os currency formatting transaction

### PR DESCRIPTION
**Pull request by aparthibanpaypal**
PR for Internal Review

### CODE REVIEW

**Review by nanbehera — APPROVED** [review]
Approved

**Review by pdamodarannair — DISMISSED** [review]

**Review by nanbehera — COMMENTED** [review]
just curious,why  Cartfile change required for currency format ?

**Comment by aparthibanpaypal** [comment]
> just curious,why Cartfile change required for currency format ?
Let me check

File: Common/Sources/Helper/Currency.swift [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+class Currency {
```
Comment (jsaini-hw) [inline Common/Sources/Helper/Currency.swift:19]: Delete this file

File: Common/Sources/Extensions/String.swift [hunk 2]
```diff
@@ -77,6 +77,37 @@ public extension String {
...
+
+        let currencyFormatter = NumberFormatter()
+        currencyFormatter.usesGroupingSeparator = true
+        currencyFormatter.numberStyle = .currency
+        currencyFormatter.currencyCode = currencyCode
+        currencyFormatter.currencySymbol = ""
+        let formattedAmountInNumber = currencyFormatter.number(from: self)
+        currencyFormatter.currencySymbol = Currency.shared.findCurrencySymbol(for: currencyCode)
+        return currencyFormatter.string(for: formattedAmountInNumber) ?? self
+    }
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/String.swift:95]: Delete this method

File: Common/Sources/Extensions/String.swift [hunk 3]
```diff
@@ -77,6 +77,37 @@ public extension String {
...
+
+    /// Format amount for currency code using users locale
+    /// - Parameter currencyCode: currency code
+    /// - Returns: a formatted amount string
+    func formatCurrencyAmount(with currencyCode: String?) -> String {
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/String.swift:100]: update func name  to `formatToCurrency `

File: Common/Tests/Extensions/StringTests.swift [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+//                     ("Uganda Currency", "1000000", "UGX", "USh 1,000,000"),
+//                     ("United States Currency", "1000000", "USD", "$1,000,000.00"),
+//                     ("Vietnam Currency", "1000000", "VND", "₫1,000,000")
+//        ]
+//        cases.forEach {
+//            XCTAssertEqual($1.formatCurrencyAmount(with: $2), $3, "\($0) test case - currency should be equal to \($3)")
+//        }
+//    }
+
+    func testFormatCurrencyAmount() {
```
Comment (jsaini-hw) [inline Common/Tests/Extensions/StringTests.swift:88]: Delete commented code and update func name to `testFormatToCurrency`. This looks good so we will go with this implementation

File: Common/Sources/Helper/Currency.swift (hunk 1 above)
Comment (aparthibanpaypal) [reply Common/Sources/Helper/Currency.swift:19]: Sure

File: Common/Sources/Extensions/String.swift (hunk 2 above)
Comment (aparthibanpaypal) [reply Common/Sources/Extensions/String.swift:95]: Okay

File: Common/Sources/Extensions/String.swift (hunk 3 above)
Comment (aparthibanpaypal) [reply Common/Sources/Extensions/String.swift:100]: Will rename it

File: Common/Tests/Extensions/StringTests.swift (hunk 4 above)
Comment (aparthibanpaypal) [reply Common/Tests/Extensions/StringTests.swift:88]: Okay

**Review by nmadshetty — DISMISSED** [review]

**Review by nanbehera — APPROVED** [review]
reviewed

**Review by pdamodarannair — DISMISSED** [review]

File: HyperwalletUISDK.xcodeproj/xcshareddata/xcschemes/Demo.xcscheme [hunk 5]
```diff
@@ -55,6 +55,11 @@
                BlueprintName = "HyperwalletUISDKUITests"
                ReferencedContainer = "container:HyperwalletUISDK.xcodeproj">
             </BuildableReference>
+            <SkippedTests>
+               <Test
+                  Identifier = "ListReceiptTests/testReceiptDetail_verifyTransactionReceiptWithOtherCurrencies()">
+               </Test>
+            </SkippedTests>
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/xcshareddata/xcschemes/Demo.xcscheme:62]: Why skip this test

File: UITests/Receipts/ListReceiptTests.swift [hunk 6]
```diff
@@ -103,6 +103,21 @@ class ListReceiptTests: BaseTests {
         XCTAssertEqual(currency, cell.staticTexts["receiptTransactionCurrencyLabel"].label)
     }
 
+    private func verifyCurrencyAndCurrencyCode(_ type: String, _ amount: String, _ currency: String, at index: Int) {
+        let cell = app.cells.element(boundBy: index)
+        app.scroll(to: cell)
+        XCTAssertTrue(cell.staticTexts["receiptTransactionTypeLabel"].exists)
+        XCTAssertTrue(cell.staticTexts["receiptTransactionAmountLabel"].exists)
+        // XCTAssertTrue(cell.staticTexts["receiptTransactionCreatedOnLabel"].exists)
```
Comment (jsaini-hw) [inline UITests/Receipts/ListReceiptTests.swift:111]: remove the commented code

Comment (aparthibanpaypal) [reply UITests/Receipts/ListReceiptTests.swift:111]: Removed the commented lines

File: HyperwalletUISDK.xcodeproj/xcshareddata/xcschemes/Demo.xcscheme (hunk 5 above)
Comment (aparthibanpaypal) [reply HyperwalletUISDK.xcodeproj/xcshareddata/xcschemes/Demo.xcscheme:62]: Added to Demo Scheme

**Review by jsaini-hw — DISMISSED** [review]

**Review by jsaini-hw — APPROVED** [review]

**Review by ddhar-hw — APPROVED** [review]

**Pull request merged**